ledger/src/blockstore.rs

```diff
@@ -2664,71 +2681,21 @@ impl Blockstore {
         }
         get_initial_slot_timer.stop();
 
-        // Check the active_transaction_status_index to see if it contains slot. If so, start with
-        // that index, as it will contain higher slots
-        let starting_primary_index = *self.active_transaction_status_index.read().unwrap();
-        let next_primary_index = u64::from(starting_primary_index == 0);
-        let next_max_slot = self
-            .transaction_status_index_cf
-            .get(next_primary_index)?
-            .unwrap()
-            .max_slot;
-
-        let mut starting_primary_index_iter_timer = Measure::start("starting_primary_index_iter");
-        if slot > next_max_slot {
-            let mut starting_iterator = self.address_signatures_cf.iter(IteratorMode::From(
-                (starting_primary_index, address, slot, Signature::default()),
+        let mut address_signatures_iter_timer = Measure::start("iter_timer");
+        let mut iterator = self
+            .address_signatures_cf
+            .iter_filtered(IteratorMode::From(
+                (address, slot, 0, Signature::default()),
```

Comment:
Lol, I have examined this logic multiple times with the same question in mind, esp when I was thinking we could support both keys in this function. I *think* I concluded that we actually want to progress to the previous slot and so this code is correct. But I will go through it with a fine-tooth comb and make sure we get it right. If it's broken, this is a fix we could pull out and PR separately, because it's pre-existing logic.

Reply:
Add comment. Edits for clarity very welcome.

Reply:
Nice, will pursue any further additions there. Also, I definitely got a bit of tunnel vision when I first made this comment 😬 :
https://github.com/solana-labs/solana/blob/ecb1f8a9d739213cca027660d74b3b090c0d5d3e/ledger/src/blockstore.rs#L2660-L2662